clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1742,14 +1923,38 @@ public void resume(Collection<TopicPartition> partitions) {
      * @return The end offsets for the given partitions.
      * @throws org.apache.kafka.common.errors.AuthenticationException if authentication fails. See the exception for more details
      * @throws org.apache.kafka.common.errors.AuthorizationException if not authorized to the topic(s). See the exception for more details
-     * @throws org.apache.kafka.common.errors.TimeoutException if the offsets could not be fetched before
-     *         expiration of the configured {@code request.timeout.ms}
      */
     @Override
     public Map<TopicPartition, Long> endOffsets(Collection<TopicPartition> partitions) {
+        return endOffsets(partitions, Duration.ofMillis(Long.MAX_VALUE));
```

Comment:
There is some internal inconsistency which is probably causing some confusion. Only `position()`, `commitSync()` and `committed()` have the indefinite blocking behavior. The rest, including this one, should use the request timeout.